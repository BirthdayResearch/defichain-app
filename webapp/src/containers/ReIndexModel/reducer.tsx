import { createSlice } from '@reduxjs/toolkit';
// TODO: Remove reducer from here
const configSlice = createSlice({
  name: 'reindexModal',
  initialState: {
    isReIndexModelOpen: false,
    isRestart: false,
  },
  reducers: {
    openReIndexModal(state) {
      state.isReIndexModelOpen = true;
    },
    closeReIndexModal(state) {
      state.isReIndexModelOpen = false;
    },
    isRestartLoader(state) {
      state.isRestart = true;
    },
    closeRestartLoader(state) {
      state.isRestart = false;
    },
  },
});

const { actions, reducer } = configSlice;

export const {
  openReIndexModal,
  closeReIndexModal,
  isRestartLoader,
  closeRestartLoader,
} = actions;

export default reducer;
