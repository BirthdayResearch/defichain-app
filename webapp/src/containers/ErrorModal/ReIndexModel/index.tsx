import React from 'react';
import { connect } from 'react-redux';
import { I18n } from 'react-redux-i18n';
import { Button, ModalHeader } from 'reactstrap';
import { Modal, ModalBody, ModalFooter } from 'reactstrap';
import { MdClose } from 'react-icons/md';
import { restartNodeWithReIndexing, closeApp } from '../../../utils/isElectron';
import { closeReIndexModal, isRestartLoader } from '../reducer';

interface ReIndexModalProps {
  isReIndexModelOpen: boolean;
  isReIndexRestart: boolean;
  closeReIndexModal: () => void;
  isRestartLoader: () => void;
}

const ReIndexModal: React.FunctionComponent<ReIndexModalProps> = (
  props: ReIndexModalProps
) => {
  const restartAppWithReIndexing = () => {
    props.closeReIndexModal();
    props.isRestartLoader();
    restartNodeWithReIndexing({ isReindexReq: true });
  };

  const closePopupAndApp = () => {
    props.closeReIndexModal();
    closeApp();
  };

  return (
    <div>
      <Modal isOpen={props.isReIndexModelOpen} centered>
        <ModalHeader>
          <MdClose onClick={props.closeReIndexModal} />
        </ModalHeader>
        <ModalBody>
          <p className='text-center'>
            {I18n.t('alerts.restartAppWithReindexNoticeline1')}
          </p>
          <p className='text-center'>
            {I18n.t('alerts.restartAppWithReindexNoticeline2')}
          </p>
        </ModalBody>
        <ModalFooter>
          <div className='d-flex justify-content-end'>
            <Button color='primary' onClick={restartAppWithReIndexing}>
              {I18n.t('alerts.yesRestartAppWithReindex')}
            </Button>
            <Button className='ml-4' onClick={closePopupAndApp}>
              {I18n.t('alerts.noCloseApp')}
            </Button>
          </div>
        </ModalFooter>
      </Modal>
    </div>
  );
};

const mapStateToProps = (state) => {
  const { isReIndexModelOpen, isReIndexRestart } = state.errorModal;

  return {
    isReIndexModelOpen,
    isReIndexRestart,
  };
};

const mapDispatchToProps = {
  closeReIndexModal,
  isRestartLoader,
};

export default connect(mapStateToProps, mapDispatchToProps)(ReIndexModal);
