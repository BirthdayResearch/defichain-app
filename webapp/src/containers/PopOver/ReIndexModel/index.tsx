import React from 'react';
import { connect } from 'react-redux';
import { Modal, ModalBody, ModalFooter, Button } from 'reactstrap';
import { restartNodeWithReIndexing, closeApp } from '../../../utils/isElectron';
import { closeReIndexModal, isRestartLoader } from '../../PopOver/reducer';
import { I18n } from 'react-redux-i18n';
import { startSetNodeVersion } from '../../RpcConfiguration/reducer';
import styles from '../popOver.module.scss';
import { MAIN } from '../../../constants';
import { snapshotDownloadRequest } from '../../SyncStatus/service';

interface ReIndexModalProps {
  isReIndexModelOpen: boolean;
  reIndexMessage?: string;
  activeNetwork: string;
  closeReIndexModal: () => void;
  isRestartLoader: () => void;
  startSetNodeVersion: () => void;
}

const ReIndexModal: React.FunctionComponent<ReIndexModalProps> = (
  props: ReIndexModalProps
) => {
  const {
    closeReIndexModal,
    isRestartLoader,
    isReIndexModelOpen,
    reIndexMessage,
    startSetNodeVersion,
    activeNetwork,
  } = props;

  const restartAppWithReIndexing = () => {
    closeReIndexModal();
    isRestartLoader();
    const params = {
      isReindexReq: true,
      skipVersionCheck: true,
      isDeletePeersAndBlocksreq: true,
    };
    startSetNodeVersion();
    restartNodeWithReIndexing(params);
  };

  const startDownloadSnapshot = async () => {
    closeReIndexModal();
    startSetNodeVersion();
    snapshotDownloadRequest();
  };

  const closePopupAndApp = () => {
    closeReIndexModal();
    closeApp();
  };

  return (
    <Modal isOpen={isReIndexModelOpen} centered className={styles.reindexModal}>
      <ModalBody>
        <h1 className='h4'>
          {I18n.t(
            !reIndexMessage
              ? 'alerts.reindexModelHeader'
              : 'alerts.nodeVersionHeader'
          )}
        </h1>

        <p>
          {reIndexMessage != null && reIndexMessage != ''
            ? reIndexMessage
            : I18n.t('alerts.restartAppWithReindexNotice')}
        </p>
      </ModalBody>
      <ModalFooter>
        <Button
          color='link'
          size='sm'
          className='mr-auto'
          onClick={closePopupAndApp}
        >
          {I18n.t('alerts.noCloseApp')}
        </Button>
        <Button size='sm' color='link' onClick={restartAppWithReIndexing}>
          {I18n.t('alerts.syncNode')}
        </Button>
        {activeNetwork === MAIN && (
          <Button size='sm' color='primary' onClick={startDownloadSnapshot}>
            {I18n.t('alerts.downloadSnapshot')}
          </Button>
        )}
      </ModalFooter>
    </Modal>
  );
};

const mapStateToProps = (state) => {
  const { isReIndexModelOpen, reIndexMessage } = state.popover;
  const { activeNetwork } = state.app;

  return {
    isReIndexModelOpen,
    reIndexMessage,
    activeNetwork,
  };
};

const mapDispatchToProps = {
  closeReIndexModal,
  isRestartLoader,
  startSetNodeVersion,
};

export default connect(mapStateToProps, mapDispatchToProps)(ReIndexModal);
