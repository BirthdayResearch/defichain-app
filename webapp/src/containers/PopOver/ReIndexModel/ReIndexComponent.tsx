import React from 'react';
import { MdClose } from 'react-icons/md';
import { connect } from 'react-redux';
import { I18n } from 'react-redux-i18n';
import { ModalHeader, ModalBody, ModalFooter, Button } from 'reactstrap';
import { closeReIndexModal } from '../../PopOver/reducer';

interface Reindexcomponent {
  restartAppWithReIndexing: () => void;
  closePopupAndApp: () => void;
  closeReIndexModal: () => void;
}

const Reindexcomponent = (props) => {
  const {
    restartAppWithReIndexing,
    closePopupAndApp,
    closeReIndexModal,
  } = props;
  return (
    <>
      <ModalHeader>
        <MdClose onClick={closeReIndexModal} />
      </ModalHeader>
      <ModalBody>
        <p className='text-center'>
          {I18n.t('alerts.restartAppWithReindexNoticeline1')}
        </p>
        <p className='text-center'>
          {I18n.t('alerts.restartAppWithReindexNoticeline2')}
        </p>
      </ModalBody>
      <ModalFooter>
        <div className='d-flex justify-content-end'>
          <Button color='primary' onClick={restartAppWithReIndexing}>
            {I18n.t('alerts.yesRestartAppWithReindex')}
          </Button>
          <Button className='ml-4' onClick={closePopupAndApp}>
            {I18n.t('alerts.noCloseApp')}
          </Button>
        </div>
      </ModalFooter>
    </>
  );
};

const mapStateToProps = () => {};

const mapDispatchToProps = {
  closeReIndexModal,
};

export default connect(mapStateToProps, mapDispatchToProps)(Reindexcomponent);
