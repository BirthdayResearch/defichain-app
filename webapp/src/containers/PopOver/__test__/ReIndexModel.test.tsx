import React from 'react';
import { Provider } from 'react-redux';
import store from '../../../app/rootStore';
import Enzyme from '../../../utils/testUtils/enzyme';
import ReIndexModel from '../ReIndexModel';

describe('Error Modal', () => {
  it('should check for snapshot', () => {
    const wrapper = Enzyme.mount(
      <Provider store={store}>
        <ReIndexModel />
      </Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
