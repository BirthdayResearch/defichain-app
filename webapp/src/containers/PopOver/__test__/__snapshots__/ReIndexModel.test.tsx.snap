// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error Modal should check for snapshot 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Connect(ReIndexModal)>
    <ReIndexModal
      closeReIndexModal={[Function]}
      isReIndexModelOpen={false}
      isRestartLoader={[Function]}
      reIndexMessage=""
      startSetNodeVersion={[Function]}
    >
      <Modal
        autoFocus={true}
        backdrop={true}
        backdropTransition={
          Object {
            "mountOnEnter": true,
            "timeout": 150,
          }
        }
        centered={true}
        container="body"
        fade={true}
        isOpen={false}
        keyboard={true}
        modalTransition={
          Object {
            "timeout": 300,
          }
        }
        onClosed={[Function]}
        onOpened={[Function]}
        returnFocusAfterClose={true}
        role="dialog"
        scrollable={false}
        trapFocus={false}
        unmountOnClose={true}
        zIndex={1050}
      />
    </ReIndexModal>
  </Connect(ReIndexModal)>
</Provider>
`;
